### PR TITLE
refactor(nextjs-template): componetize the parts of DendronLayout.tsx

### PR DESCRIPTION
**Pull request by hkievet**
In addition to cleaning up DendronLayout, this also optimizes Mermaid so that it only loads the mermaid library once and only once, when a note page that has a mermaid chart is encountered.

# Cleanup some of the frontend code
- Track if isMobile and whether or not the sidebar is open using a useDendronContext hook.  Can add more to that state in the future.
- Split the DendronLayout component to make it easier to grok in the future.

# Dendron Extended PR Checklist

- NOTE: the links don't work. you'll need to go into the wiki and use lookup to find the note until we fix some issues in the markdown export

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [ ] check whether code be simplified
  - [ ] check if similar function already exist in the codebase. if so, can it be re-used?
  - [ ] check if this change adversely impact performance
- Operations
  - [ ] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [ ] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [ ] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [ ] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [ ] [Confirm manual testing](dev.process.qa.test) 
- [ ] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended
- [ ] If you are adding a new language feature (graphically visible in VS Code/preview/publishing), an example is included in the [test workspace](dev.ref.test-workspace)
- CSS
  - [x] display is correct for following dimensions
    - [x] sm: screen ≥ 576px, eg. iphonex, (375x812)
    - [x] lg: screen ≥ 992px
    - [ ] xxl: screen ≥ 1600px eg. mac (1600x900)
  - [ ] display is correct for following browsers (across the various dimensions)
    - [ ] safari
    - [x] firefox
    - [x] chrome

## Docs
- [ ] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [ ] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [ ]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

**Comment by kevinslin** [comment]
> [CLA assistant check](https://cla-assistant.io/dendronhq/dendron?pullRequest=3336) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/dendronhq/dendron?pullRequest=3336) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ hkievet❌ [hkievet@gmail.com](mailto:hkievet@gmail.com)
> **[hkievet@gmail.com](mailto:hkievet@gmail.com)** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dendronhq/dendron?pullRequest=3336) it.

looks like you need to add your gmail to github to sign the CLA

File: packages/nextjs-template/context/DendronProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { convertLegacyProps } from "antd/lib/button/button";
```
Comment (kevinslin) [inline packages/nextjs-template/context/DendronProvider.tsx:1]: remove?

File: packages/nextjs-template/context/useDendronContext.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { useContext } from "react";
+import { dendronContext } from "./DendronProvider";
+
+export function useDendronContext() {
```
Comment (kevinslin) [inline packages/nextjs-template/context/useDendronContext.tsx:4]: not against providing our own context but pros/cons of using this vs a redux slice?

**Review by kevinslin — CHANGES_REQUESTED** [review]
nice job. only some minor comments. excited to get this merged :)

**Comment by hkievet** [comment]
@kevinslin When I press "recheck" it just brings me to the homepage of github.  I updated my email so now it's "public", which I'm not a huge fan of, but I guess it's not that big of a secret anyways haha.  I'm not sure how to move forward with the CLA agreement issues.

File: packages/nextjs-template/context/useDendronContext.tsx (hunk 2 above)
Comment (hkievet) [reply packages/nextjs-template/context/useDendronContext.tsx:4]: Basic Use Context:
Pros
+ Simple
+ TypeScript Friendly
+ Can easily switch to Redux if state management becomes more tedious

Cons
- Can't use Redux middleware


Redux Slice:
Pros:
+ Redux middlewares can sometimes provide extra benefit

Cons:
- Lots of boilerplate
- Redux requires a lot of extra TypeScript

File: packages/nextjs-template/context/DendronProvider.tsx (hunk 1 above)
Comment (hkievet) [reply packages/nextjs-template/context/DendronProvider.tsx]: Removed with a force push amend to the mermaid commit.

**Review by kevinslin — APPROVED** [review]
